### ISSUE TITLE
Invalid read of size 4 in jxl::PerformBlending

### ISSUE THREAD

**Issue by novomesk** [libjxl/libjxl#1736]
[Invalid_read_of_size_4.zip](https://github.com/libjxl/libjxl/files/9542867/Invalid_read_of_size_4.zip)

Testing latest commit from [0.7.x](https://github.com/libjxl/libjxl/tree/v0.7.x) branch.

`valgrind djxl clusterfuzz-testcase-minimized-kimgio_jxl_fuzzer-6272022027567104.jxl out.png`

```
==17945== Invalid read of size 4
==17945==    at 0x49A2DE0: jxl::PerformBlending(float const* const*, float const* const*, float* const*, unsigned long, unsigned long, jxl::PatchBlending const&, jxl::PatchBlending const*, std::vector<jxl::ExtraChannelInfo, std::allocator<jxl::ExtraChannelInfo> > const&) (in /usr/lib64/libjxl.so.0.7.0)
==17945==    by 0x4A6303E: jxl::N_AVX2::BlendingStage::ProcessRow(std::vector<std::vector<float*, std::allocator<float*> >, std::allocator<std::vector<float*, std::allocator<float*> > > > const&, std::vector<std::vector<float*, std::allocator<float*> >, std::allocator<std::vector<float*, std::allocator<float*> > > > const&, unsigned long, unsigned long, unsigned long, unsigned long, unsigned long) const (in /usr/lib64/libjxl.so.0.7.0)
==17945==    by 0x4A5C0AA: jxl::LowMemoryRenderPipeline::RenderRect(unsigned long, std::vector<jxl::Plane<float>, std::allocator<jxl::Plane<float> > >&, jxl::RectT<unsigned long>, jxl::RectT<unsigned long>) (in /usr/lib64/libjxl.so.0.7.0)
==17945==    by 0x4A5CE5B: jxl::LowMemoryRenderPipeline::ProcessBuffers(unsigned long, unsigned long) (in /usr/lib64/libjxl.so.0.7.0)
==17945==    by 0x49FA0B1: jxl::ThreadPool::RunCallState<jxl::ModularFrameDecoder::FinalizeDecoding(jxl::PassesDecoderState*, jxl::ThreadPool*, bool)::{lambda(unsigned long)#1}, jxl::ModularFrameDecoder::FinalizeDecoding(jxl::PassesDecoderState*, jxl::ThreadPool*, bool)::{lambda(unsigned int, unsigned long)#2}>::CallDataFunc(void*, unsigned int, unsigned long) (in /usr/lib64/libjxl.so.0.7.0)
==17945==    by 0x49A268B: jxl::ThreadPool::SequentialRunnerStatic(void*, void*, int (*)(void*, unsigned long), void (*)(void*, unsigned int, unsigned long), unsigned int, unsigned int) (in /usr/lib64/libjxl.so.0.7.0)
==17945==    by 0x49FA54D: jxl::ModularFrameDecoder::FinalizeDecoding(jxl::PassesDecoderState*, jxl::ThreadPool*, bool) (in /usr/lib64/libjxl.so.0.7.0)
==17945==    by 0x49D8D4E: jxl::FrameDecoder::Flush() (in /usr/lib64/libjxl.so.0.7.0)
==17945==    by 0x49D9024: jxl::FrameDecoder::FinalizeFrame() (in /usr/lib64/libjxl.so.0.7.0)
==17945==    by 0x4A0B6E6: JxlDecoderProcessInput (in /usr/lib64/libjxl.so.0.7.0)
==17945==    by 0x117587: jxl::extras::DecodeImageJXL(unsigned char const*, unsigned long, jxl::extras::JXLDecompressParams const&, unsigned long*, jxl::extras::PackedPixelFile*, std::vector<unsigned char, std::allocator<unsigned char> >*) (in /usr/bin/djxl)
==17945==    by 0x110BC4: main (in /usr/bin/djxl)
```

**Comment by mo271** [comment]
This was introduced by https://github.com/libjxl/libjxl/commit/a426a680.
In 
https://github.com/libjxl/libjxl/blob/137f53ef1a6f39e5f733edb6cc457625abb3b3c9/lib/jxl/render_pipeline/stage_blending.cc#L146
there is a silent assumption that the size of `ec_bg` is the same as `bg_`. However in this fuzzer image, this is not the case:
`bg_` has size 8x8 while for the failing extra channel, `ec_bg` has size 4x4 and then it fails when trying to access row 4 in this extra channel.
Not sure how to fix that best. Is it allowed for the extra channels to have different sizes when blending? If not, perhaps we should check that somewhere?
What do you think, @veluca93 and @sboukortt?

**Comment by mo271** [comment]
This has been fixed by #1838
